index.js

```diff
@@ -0,0 +1,186 @@
+    for (let key in itemData) {
+        if (typeof itemData[key] === 'string') {
+            const [categoryName, categoryValue] = textForListItem(itemData, key);
+
+            const listItem = createElem('li', 'item__list-item', categoryName);
+            const span = createElem('span', '', categoryValue);
+
+            listItem.append(span);
+            itemList.append(listItem);
+        }
+    }
+
+    itemWrap.addEventListener('click', openCard.bind(null, itemWrap, itemList));
+    return itemWrap;
+}
+
+function textForListItem(data, key) {
+    for (let i = 0; i < forbiddenKeys.length; i++) {
+        if (key == forbiddenKeys[i]) {
+            return '';
+        }
+    }
+    return convertText(data, key);
+}
+
+function convertText(data, key) {
+    let [firstPart, secondPart] = key.split('_');
+    firstPart = firstPart.charAt(0).toUpperCase() + firstPart.slice(1);
+    const categoryName = `${firstPart}${secondPart ? ` ${secondPart}` : ''}: `;
+
+return [categoryName, data[key]];
+}
+
+function openCard(itemWrap, itemList) {
+  if (itemList.classList.contains('open')) {
+      itemList.classList.remove('open');
+      itemList.style.height = '';
+      itemWrap.style.alignSelf = '';
+  } else {
+      itemList.classList.add('open');
+      itemWrap.classList.add('open');
+      itemList.style.height = itemList.scrollHeight + 'px';
+  }
+}
+
+function createElem(elemName = 'div', nameOfClass = '', text = '') {
+  const elem = document.createElement(elemName);
+  elem.className = nameOfClass;
+  elem.textContent = text;
+  return elem;
+}
+
+function addNavLinksListeners() {
+  const navLinks = navList.querySelectorAll('.nav__category');
+  navLinks.forEach(item => {
+      item.addEventListener('click', clickHandler);
+  })
+}
+
+function renderPagination(href, countOfPagination) {
+  clearPagination();
+  if (countOfPagination > 1) {
+      const paginationList = createElem('ul', 'pagination__list');
+      for (let i = 0; i < countOfPagination; i++) {
+          const paginationItem = createElem('li', 'pagination__item');
+          const paginationLink = createPaginationLink(href, i + 1);
+          paginationLink.addEventListener('click', clickHandler);
+          paginationItem.prepend(paginationLink);
+          paginationList.append(paginationItem);
+      }
+      pagination.prepend(paginationList);
+  }
+}
+
+function clearPagination() {
+  const clearPaginationList = document.querySelector('.pagination__list');
+  if (clearPaginationList) {
+      const oldLinks = clearPaginationList.querySelectorAll('a');
+      oldLinks.forEach(item => item.removeEventListener('click', clickHandler));
+      clearPaginationList.remove();
+  }
+}
+function setImageBackgroundUrl() {
+  const ibg = document.querySelectorAll(".ibg");
+  for (let i = 0; i < ibg.length; i++) {
+      const img = ibg[i].querySelector('.img-ibg');
+
+      if (img) {
+          ibg[i].style.backgroundImage = `url(${img.src})`;
+      };
+  };
+}

```

Comment:
Absolutely unnecessary function. Set the background with HTML or CSS.

Reply:
I saw this code somewhere in freelance videos on youtube, I can even feel it 💯